File: android/UClass/app/src/main/java/com/ubase/uclass/network/response/SNSLoginData.kt
package com.ubase.uclass.network.response

import com.google.gson.annotations.SerializedName

/**
 * SNS 로그인 응답 데이터 모델
 */
data class SNSLoginData(
    @SerializedName("userId")
    val userId: Int,

    @SerializedName("approvalStatus")
    val approvalStatus: String,

    @SerializedName("userName")
    val userName: String,

    @SerializedName("branchName")
    val branchName: String,

    @SerializedName("tokenType")
    val tokenType: String,

    @SerializedName("loginAt")
    val loginAt: String,

    @SerializedName("accessToken")
    val accessToken: String,

    @SerializedName("expiresIn")
    val expiresIn: Int,

    @SerializedName("userType")
    val userType: String,

    @SerializedName("branchId")
    val branchId: Int,

    @SerializedName("redirectUrl")
    val redirectUrl: String,

    @SerializedName("reasonUrl")
    val reasonUrl: String
)

File: ios/uclass/uclass/network/response/SNSLoginData.swift
import Foundation

// MARK: - SNS 로그인 응답 데이터 모델
struct SNSLoginData: Codable {
    // ✅ 옵셔널로 변경 - 승인 대기 중에는 일부 필드만 존재
    let userId: Int?
    let approvalStatus: String?
    let userName: String?
    let branchName: String?
    let tokenType: String?
    let loginAt: String?
    let accessToken: String?
    let expiresIn: Int?
    let userType: String?
    let branchId: Int?
    let redirectUrl: String?
    let reasonUrl: String?
    
    enum CodingKeys: String, CodingKey {
        case userId
        case approvalStatus
        case userName
        case branchName
        case tokenType
        case loginAt
        case accessToken
        case expiresIn
        case userType
        case branchId
        case redirectUrl
        case reasonUrl
    }
}
